streams-contrib/streams-provider-twitter/src/main/java/org/apache/streams/twitter/api/AccountActivity.java

```diff
@@ -0,0 +1,127 @@
+
+  /**
+   * Registers a new webhook URL for the given application context. The URL will be validated via CRC request before saving. In case the validation fails, a comprehensive error is returned. message to the requester.
+   *
+   * Only one webhook URL can be registered to an application.
+   *
+   * @return Webhook
+   * @see <a href=https://dev.twitter.com/webhooks/reference/post/account_activity/webhooks">https://dev.twitter.com/webhooks/reference/post/account_activity/webhooks</a>
+   *
+   */
+  @RemoteMethod(httpMethod = "POST", path = "/webhooks.json")
+  public Webhook registerWebhook(@Query("url") String url);
+
+  /**
+   * Removes the webhook from the provided application’s configuration. The webhook ID can be accessed by making a call to GET /1.1/account_activity/webhooks.
+   *
+   * @return Boolean
+   * @see <a href=https://dev.twitter.com/webhooks/reference/del/account_activity/webhooks">https://dev.twitter.com/webhooks/reference/del/account_activity/webhooks</a>
+   *
+   */
+  @RemoteMethod(httpMethod = "DELETE", path = "/webhooks/{webhookId}.json")
+  public Boolean deleteWebhook(@Path("webhookId") Long webhookId);
+
+  /**
+   * Triggers the challenge response check (CRC) for the given webhook’s URL. If the check is successful, returns 204 and reenables the webhook by setting its status to valid.
+   *
+   * @param webhookId Webhook ID. Defined in resource path.
+   * @return Boolean
+   * @see <a href=https://dev.twitter.com/webhooks/reference/put/account_activity/webhooks">https://dev.twitter.com/webhooks/reference/put/account_activity/webhooks</a>
+   *
+   */
+  @RemoteMethod(httpMethod = "PUT", path = "/webhooks/{webhookId}.json")
+  public Boolean putWebhook(@Path("webhookId") Long webhookId);
+
+  /**
+   * Provides a way to determine if a webhook configuration is subscribed to the provided user’s Direct Messages. If the provided user context has an active subscription with the provided app, returns 204 OK. If the response code is not 204, then the user does not have an active subscription. See HTTP Response code and error messages below for details.
+   *
+   * @param webhookId Webhook ID. Defined in resource path.
+   * @return Boolean
+   * @see <a href=https://dev.twitter.com/webhooks/reference/put/account_activity/webhooks">https://dev.twitter.com/webhooks/reference/put/account_activity/webhooks</a>
+   *
+   */
+  @RemoteMethod(httpMethod = "GET", path = "/webhooks/{webhookId}/subscriptions.json")
+  public Boolean getWebhookSubscription(@Path("webhookId") Long webhookId)
+      throws InvocationTargetException, RestCallException
+  ;
+
+  /**
+   * Subscribes the provided app to events for the provided user context. When subscribed, all DM events for the provided user will be sent to the app’s webhook via POST request.
+   *
+   * Subscriptions are currently limited to 50 users per application. If you have a need to add more subscriptions, please let us know by using this form.
```

Comment:
This part of the docs is a little weird. Should add https://gnipinc.formstack.com/forms/account_activity_api_elevation_request_form to it so it links to something.

Reply:
on second thought, the javadocs don't need that line.